files/class-meta-updater.php

```diff
@@ -175,7 +175,7 @@ private function update_attachment_filesize( $attachment_id ): array {
 
 		$filesize = $this->get_filesize_from_file( $attachment_id );
 
-		if ( 0 >= $filesize ) {
+		if ( $filesize < 0 ) {
```

Comment:
The code was updated to return 0 only for 404 errors. Or when the attachment is genuinely 0 bytes. Fail only in the case of other failures. Supposedly, this will allow us to avoid infinite loops.